blog/2025-06-openapi-generator/README.md

```diff
@@ -0,0 +1,465 @@
+
+<hr>
+
+## Why you need a code generator for your Angular API clients
+
+Every Angular project that talks to a backend faces the same repetitive pain: writing boilerplate code to interact with REST endpoints. 
+HTTP clients, models, DTOs, error handling, and keeping everything in sync when the API changes - it adds up fast and slows teams down.
+
+Wouldn't it be great if all of that could be automated?
+
+That's exactly what **API code generators** do. 
+They take your OpenAPI specification and generate fully-typed, ready-to-use client code: consistently, reliably, and within seconds. 
+No more hand-written HTTP services. 
+No more copy-pasting types. 
+No more drift between frontend and backend.
+
+A few years ago, I wrote a popular article on [using Swagger Codegen to generate Angular clients](/blog/2018-04-swagger-codegen). 
+Since then, the ecosystem has moved forward: **OpenAPI Generator** emerged as the actively maintained, community-driven successor to Swagger Codegen. 
+It's more powerful, better supported, and trusted by companies and open-source projects around the world.
+
+> 👉 In this updated article, I'll show you how easy it is to generate a complete API client for Angular using OpenAPI Generator - and how to integrate it seamlessly into your Angular project.
+
+**Let's get started! 🚀**
+
+
+## What you need
+
+- A valid OpenAPI 2.0 or 3.x spec file (JSON/YAML) that describes your REST backend.
+- Java 11+ installed, or Docker installed (if installing Java directly is not an option)
+
+
+## Installation: Use the CLI wrapper for consistent builds
+
+Theoretically, you could download the latest version of the code generator manually: 
+just grab the `openapi-generator-cli.jar` from [Maven Central](https://repo1.maven.org/maven2/org/openapitools/openapi-generator-cli/).
+
+But in practice, I **highly recommend** using the official node.js-based CLI wrapper instead.
+
+```bash
+npm install @openapitools/openapi-generator-cli --save-dev
+```
+
+This small package gives you a simple command-line tool: `openapi-generator-cli`.
+Once installed, you can execute it via `npx openapi-generator-cli`.
+It gives you the look and feel of all the other node.js based tooling that you are already used to.
+But more importantly, it installs a **specific version** of the generator and writes that information into the file `openapitools.json` inside your project.
+
+Now, when you check your project into Git (including the file `openapitools.json`), your teammates - or the CI pipeline - will automatically use **the exact same version** when they run the CLI.
+
+- This ensures **100% reproducible code generation** across your entire team.
+- No "works on my machine" issues due to different generator versions.
+- No unexpected diffs caused by subtle changes in templates or behavior.
+
+That's exactly what we want for a clean and reliable workflow.
+
+### Switching Code Generator Versions
+
+Let's say you want to lock your project to version `7.13.0`. 

```

Comment:
a bit more elaborate so that the section headline doesn't have to be read to understand what you want to say.